sql/parser/datum.go

```diff
@@ -237,6 +237,11 @@ func (d DFloat) IsMin() bool {
 }
 
 func (d DFloat) String() string {
+	if _, frac := math.Modf(float64(d)); frac == 0 {
+		// d is a whole number, so one digit of decimal precision is included
+		// to ensure that it will parse as a float.
+		return strconv.FormatFloat(float64(d), 'f',  1, 64)
+	}
```

Comment:
Note that large whole numbers will no longer print with exponents.
For example: 1,000,000 previously: 1e+06 now: 1000000.0